colink/protocol.py

```diff
@@ -102,7 +99,11 @@ def run(
                         break
                 else:
                     err = q.get()
-                    raise err
+                    # in instance server and run_attach mode+standalone MQ, server closing MQ when shutdown may trigger this exception
```

Comment:
Based on what you wrote in `run_attach`, it seems that line: 102-106 would never happen?